tools/licenses/lib/licenses.dart

```diff
@@ -923,22 +921,22 @@ Iterable<_LicenseMatch> _tryReferenceByType(String body, RegExp pattern, License
   }
 }
 
-License _dereferenceLicense(int groupIndex, String Function(int index) group, MultipleVersionedLicenseReferencePattern pattern, LicenseSource parentDirectory, { String origin }) {
-  License result = pattern.checkLocalFirst ? parentDirectory.nearestLicenseWithName(group(groupIndex)) : null;
+License _dereferenceLicense(int groupIndex, String? Function(int? index) group, MultipleVersionedLicenseReferencePattern pattern, LicenseSource parentDirectory, { String? origin }) {
```

Comment:
i think `String? Function(int index) group` would be closer to the original intent. requires some `!`s below but removes some ugly casts later.

Reply:
Done